Libraries/Opc.Ua.Server/Configuration/ConfigurationNodeManager.cs

```diff
@@ -320,10 +320,9 @@ public void HasApplicationSecureAdminAccess(ISystemContext context, string _)
                     throw new ServiceResultException(StatusCodes.BadUserAccessDenied, "Secure Application Administrator access required.");
                 }
 
-                // allow access to system configuration only through special identity
-                IUserIdentity user = context.UserIdentity as RoleBasedIdentity;               
-                if (user == null || user.TokenType == UserTokenType.Anonymous ||
-                    !user.GrantedRoleIds.Contains(ObjectIds.WellKnownRole_SecurityAdmin))
+                // allow access to system configuration only with Role SecurityAdmin
```

Comment:
in this case I'd prefer the old implementation because it is safer. The null check is done on the same variable that is later used in the compares